lib/SPIRV/SPIRVToOCL20.cpp

```diff
@@ -211,350 +97,32 @@ void SPIRVToOCL20::visitCallSPIRVMemoryBarrier(CallInst *CI) {
+          }
         }
-        return DemangledName;
+        return CI->getName();
       },
       &Attrs);
-}
-
-void SPIRVToOCL20::translateMangledAtomicTypeName() {
-  for (auto &I : M->functions()) {
-    if (!I.hasName())
-      continue;
-    std::string MangledName = I.getName();
-    std::string DemangledName;
-    if (!oclIsBuiltin(MangledName, &DemangledName) ||
-        DemangledName.find(kOCLBuiltinName::AtomPrefix) != 0)
-      continue;
-    auto Loc = MangledName.find(kOCLBuiltinName::AtomPrefix);
-    Loc = MangledName.find(kMangledName::AtomicPrefixInternal, Loc);
-    MangledName.replace(Loc, strlen(kMangledName::AtomicPrefixInternal),
-                        MangledAtomicTypeNamePrefix);
-    I.setName(MangledName);
-  }
-}
-
-std::string SPIRVToOCL20::getGroupBuiltinPrefix(CallInst *CI) {
-  std::string Prefix;
-  auto ES = getArgAsScope(CI, 0);
-  switch (ES) {
-  case ScopeWorkgroup:
-    Prefix = kOCLBuiltinName::WorkPrefix;
-    break;
-  case ScopeSubgroup:
-    Prefix = kOCLBuiltinName::SubPrefix;
-    break;
-  default:
-    llvm_unreachable("Invalid execution scope");
-  }
-  return Prefix;
-}
-
-void SPIRVToOCL20::visitCastInst(CastInst &Cast) {
-  if (!isa<ZExtInst>(Cast) && !isa<SExtInst>(Cast) && !isa<TruncInst>(Cast) &&
-      !isa<FPTruncInst>(Cast) && !isa<FPExtInst>(Cast) &&
-      !isa<FPToUIInst>(Cast) && !isa<FPToSIInst>(Cast) &&
-      !isa<UIToFPInst>(Cast) && !isa<SIToFPInst>(Cast))
-    return;
-
-  Type const *SrcTy = Cast.getSrcTy();
-  Type *DstVecTy = Cast.getDestTy();
-  // Leave scalar casts as is. Skip boolean vector casts becase there
-  // are no suitable OCL built-ins.
-  if (!DstVecTy->isVectorTy() || SrcTy->getScalarSizeInBits() == 1 ||
-      DstVecTy->getScalarSizeInBits() == 1)
-    return;
-
-  // Assemble built-in name -> convert_gentypeN
-  std::string CastBuiltInName(kOCLBuiltinName::ConvertPrefix);
-  // Check if this is 'floating point -> unsigned integer' cast
-  CastBuiltInName += mapLLVMTypeToOCLType(DstVecTy, !isa<FPToUIInst>(Cast));
-
-  // Replace LLVM conversion instruction with call to conversion built-in
-  BuiltinFuncMangleInfo Mangle;
-  // It does matter if the source is unsigned integer or not. SExt is for
-  // signed source, ZExt and UIToFPInst are for unsigned source.
-  if (isa<ZExtInst>(Cast) || isa<UIToFPInst>(Cast))
-    Mangle.addUnsignedArg(0);
-
-  AttributeList Attributes;
-  CallInst *Call =
-      addCallInst(M, CastBuiltInName, DstVecTy, Cast.getOperand(0), &Attributes,
-                  &Cast, &Mangle, Cast.getName(), false);
-  Cast.replaceAllUsesWith(Call);
-  Cast.eraseFromParent();
+  return SPIRVToOCL::visitCallSPIRVAtomicBuiltin(CiFixedAs, OC);
 }
 
 } // namespace SPIRV
 
 INITIALIZE_PASS(SPIRVToOCL20, "spvtoocl20",
                 "Translate SPIR-V builtins to OCL 2.0 builtins", false, false)
 
-ModulePass *llvm::createSPIRVToOCL20() { return new SPIRVToOCL20(); }
+ModulePass *llvm::createSPIRVToOCL20() { return new SPIRVToOCL20(); }
```

Comment:
GitHub is still showing that files are not ending with a newline, please fix before merging.

Reply:
Fixed.